src/peer_id.rs

```diff
@@ -16,32 +17,32 @@ pub struct PeerId {
     public_key: PublicKey,
 }
 
-impl PeerId {
-    /// Create a new PeerId from a public key
-    pub fn from_public_key(public_key: PublicKey) -> PeerId {
-        PeerId { public_key }
-    }
+#[derive(Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord)]
+pub struct PeerNetID<T>(pub T)
+where
+    T: PartialEq + Eq + HashTrait + PeerNetIdTrait;
+
+pub trait PeerNetIdTrait: PartialEq + Eq + HashTrait {
```

Comment:
I think having "trait" in the naming is a bit too much

Reply:
maybe there is a better way but we need to make sure that the ID implement PartialEq + Eq + Hash and (from_bytes / verify_signature)